### PR TITLE
8344181: Remove SecurityManager and related calls from jdk.management and jdk.management.agent

### PR DESCRIPTION
**Pull request by kevinjwalls**
Remove redundant SecurityManager, AccessController references
(following on from JDK-8338411: Implement JEP 486: Permanently Disable the Security Manager).

src/jdk.management/share/classes/com/sun/management/internal/GarbageCollectionNotifInfoCompositeData.java
There is an existing theoretical path where GcInfoBuilder stays null, should never happen, "com.sun.management.GcInfo" exists...

src/jdk.management/share/classes/com/sun/management/internal/GcInfoCompositeData.java
Similarly there is an existing assumption that Class.forName("com.sun.management.GcInfo") succeeds.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344181](https://bugs.openjdk.org/browse/JDK-8344181): Remove SecurityManager and related calls from jdk.management and jdk.management.agent (**Enhancement** - P4)

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**) Review applies to [84d59ac3](https://git.openjdk.org/jdk/pull/22155/files/84d59ac39365167792e9d66fe81e57150af837f0)
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22155/head:pull/22155` \
`$ git checkout pull/22155`

Update a local copy of the PR: \
`$ git checkout pull/22155` \
`$ git pull https://git.openjdk.org/jdk.git pull/22155/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22155`

View PR using the GUI difftool: \
`$ git pr show -t 22155`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22155.diff">https://git.openjdk.org/jdk/pull/22155.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22155#issuecomment-2482729073)
</details>

### CODE REVIEW

File: src/jdk.management/share/classes/com/sun/management/internal/GarbageCollectionNotifInfoCompositeData.java [hunk 1]
```diff
@@ -65,19 +63,17 @@ public static CompositeData toCompositeData(GarbageCollectionNotificationInfo in
...
-                    }
-                }
-            });
+
+        GcInfoBuilder builder = null;
+        try {
+            Class<?> cl = Class.forName("com.sun.management.GcInfo");
+            Field f = cl.getDeclaredField("builder");
+            f.setAccessible(true);
+            builder = (GcInfoBuilder) f.get(gcNotifInfo.getGcInfo());
```
Comment (alexmenkov) [inline src/jdk.management/share/classes/com/sun/management/internal/GarbageCollectionNotifInfoCompositeData.java:72]:
```suggestion
            builder = (GcInfoBuilder)f.get(gcNotifInfo.getGcInfo());
```

File: src/jdk.management/share/classes/com/sun/management/internal/GcInfoCompositeData.java [hunk 2]
```diff
@@ -71,32 +69,24 @@ public GcInfo getGcInfo() {
...
-                                return null;
-                            }
-                        }
-                    });
+        GcInfoBuilder builder = null;
+        try {
+            Class<?> cl = Class.forName("com.sun.management.GcInfo");
+            Field f = cl.getDeclaredField("builder");
+            f.setAccessible(true);
+            builder =  (GcInfoBuilder)f.get(info);
```
Comment (alexmenkov) [inline src/jdk.management/share/classes/com/sun/management/internal/GcInfoCompositeData.java:77]:
```suggestion
            builder = (GcInfoBuilder)f.get(info);
```

File: src/jdk.management/share/classes/com/sun/management/internal/HotSpotDiagnostic.java [hunk 3]
```diff
@@ -53,20 +49,11 @@ public HotSpotDiagnostic() {
     public void dumpHeap(String outputFile, boolean live) throws IOException {
 
         String propertyName = "jdk.management.heapdump.allowAnyFileSuffix";
-        PrivilegedAction<Boolean> pa = () -> Boolean.parseBoolean(System.getProperty(propertyName, "false"));
-        @SuppressWarnings("removal")
-        boolean allowAnyFileSuffix = AccessController.doPrivileged(pa);
+        boolean allowAnyFileSuffix = Boolean.parseBoolean(System.getProperty(propertyName, "false"));
```
Comment (alexmenkov) [inline src/jdk.management/share/classes/com/sun/management/internal/HotSpotDiagnostic.java:52]: can be simplified to `Boolean.getBoolean(propertyName)`

Comment (kevinjwalls) [reply src/jdk.management/share/classes/com/sun/management/internal/HotSpotDiagnostic.java]: Yes, missed that.

File: src/jdk.management/share/classes/com/sun/management/internal/GcInfoCompositeData.java (hunk 2 above)
Comment (kevinjwalls) [reply src/jdk.management/share/classes/com/sun/management/internal/GcInfoCompositeData.java]: done

File: src/jdk.management/share/classes/com/sun/management/internal/GarbageCollectionNotifInfoCompositeData.java (hunk 1 above)
Comment (kevinjwalls) [reply src/jdk.management/share/classes/com/sun/management/internal/GarbageCollectionNotifInfoCompositeData.java:72]: OK, it was missing the space before, when moving it around I added the space, as that's what we normally have.  In this directory there is a mixture of the two styles, with and without spaces.  I hope it's OK to let this one migrate to the more modern style. 8-)

File: src/jdk.management.agent/share/classes/jdk/internal/agent/Agent.java [hunk 4]
```diff
@@ -36,7 +36,6 @@
 import java.net.MalformedURLException;
 import java.net.UnknownHostException;
 import java.nio.BufferUnderflowException;
-import java.security.AccessController;
 import java.security.PrivilegedAction;
```
Comment (plummercj) [inline src/jdk.management.agent/share/classes/jdk/internal/agent/Agent.java:39]: It doesn't look like java.security.PrivilegedAction is needed.

File: src/jdk.management/share/classes/com/sun/management/internal/DiagnosticCommandImpl.java [hunk 5]
```diff
@@ -149,13 +149,6 @@ private class Wrapper {
         }
 
         public String execute(String[] args) {
-            if (permission != null) {
```
Comment (plummercj) [inline src/jdk.management/share/classes/com/sun/management/internal/DiagnosticCommandImpl.java:152]: There is a lot of code that remains related to assigning a value to `permission`. It looks like it should also be removed.

File: src/jdk.management/share/classes/com/sun/management/internal/HotSpotDiagnostic.java [hunk 6]
```diff
@@ -167,27 +153,8 @@ public void dumpThreads(String outputFile, ThreadDumpFormat format) throws IOExc
...
-            };
-            try {
-                AccessController.doPrivileged(pa);
-            } catch (PrivilegedActionException pae) {
-                Throwable cause = pae.getCause();
-                if (cause instanceof IOException ioe)
-                    throw ioe;
-                if (cause instanceof RuntimeException e)
-                    throw e;
-                throw new RuntimeException(cause);
```
Comment (plummercj) [inline src/jdk.management/share/classes/com/sun/management/internal/HotSpotDiagnostic.java:189]: I think you've made an assumption that these are all related to lack of prividleges. I don't think that is the case, and they are all exceptions that can result from calling dumpThreads(). I think you need a try block around the dumpThreads() call that handles all these exceptions in the same manner.

**Review by plummercj — CHANGES_REQUESTED** [review]

File: src/jdk.management/share/classes/com/sun/management/internal/DiagnosticCommandImpl.java (hunk 5 above)
Comment (kevinjwalls) [reply src/jdk.management/share/classes/com/sun/management/internal/DiagnosticCommandImpl.java:152]: Right, Permission in the class Wrapper, commandDescriptor manipulates a map with permission info, there's native code that passes this between the JVM and library code....
Would like to do that separately in:
JDK-8334493: Remove SecurityManager Permissions infrastructure from DiagnosticCommands

File: src/jdk.management/share/classes/com/sun/management/internal/HotSpotDiagnostic.java (hunk 6 above)
Comment (kevinjwalls) [reply src/jdk.management/share/classes/com/sun/management/internal/HotSpotDiagnostic.java:189]: OK yes it looks that way but I think we're OK:

Previously, IOException and RuntimeException were rethrown, and that is I think everything we expected to see wrapped in an PrivilegedActionException.
Anything else was wrapped in a RuntimeException, but outside of IOException which was declared to be thrown, surely it's going to be an unchecked RuntimeException...


Here in dumpThreads we declare IOException and Runtime is unchecked.
So those two get thrown as before, not sure if we need to catch anything else.


Experimenting: invoking dumpHeap through the HotSpotDiagnostic MBean, I can see:

javax.management.RuntimeMBeanException: java.lang.IllegalArgumentException: heapdump file must have .hprof extension
or
javax.management.MBeanException: java.io.IOException: Permission denied


It looks like IO or IAException (which is a RuntimeException) get handled OK, and the MBean invocation, which called dumpHeap, wraps them in other exceptions.

File: src/jdk.management.agent/share/classes/jdk/internal/agent/Agent.java (hunk 4 above)
Comment (kevinjwalls) [reply src/jdk.management.agent/share/classes/jdk/internal/agent/Agent.java:39]: thanks, and found one more in PlatformMBeanProviderImpl.java, will add it to the PR.

**Review by plummercj — APPROVED** [review]

File: src/jdk.management/share/classes/com/sun/management/internal/VirtualThreadSchedulerImpls.java [hunk 7]
```diff
@@ -59,7 +59,6 @@ private abstract static class BaseVirtualThreadSchedulerImpl
 
         @Override
         public final void setParallelism(int size) {
-            Util.checkControlAccess();
```
[the remark this answers is not in the document]
Comment (kevinjwalls) [reply src/jdk.management/share/classes/com/sun/management/internal/VirtualThreadSchedulerImpls.java:62]: Got it, yes can remove implSetParallelism.

**Review by alexmenkov — APPROVED** [review]

**Comment by kevinjwalls** [comment]
Thanks for the reviews!

**Pull request closed without merging**